src/service/user.service.js

```diff
@@ -0,0 +1,65 @@
+import { ApiError } from '../exceptions/api.error.js';
+import { User } from '../modules/User.js';
+import { Console } from '../utils/Console.js';
+
+function validateEmail(value) {
+  const EMAIL_PATTERN = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!value) {
+    return 'Email is required';
+  }
+
+  if (!EMAIL_PATTERN.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+async function validUserName(value) {
+  const existUser = await User.findOne({ where: { value } });
```

Comment:
The `validUserName` function should check for the existence of a user by `userName` instead of `value`. Update the query to `{ where: { userName: value } }`.